geowebcache/core/src/main/java/org/geowebcache/GeoWebCacheTypedEnvironment.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * <p>You should have received a copy of the GNU Lesser General Public License along with this
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Fernando Mino, GeoSolutions S.A.S., Copyright 2019
+ */
+package org.geowebcache;
+
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+
+/** Enviroment variables placeholder resolver with declared types. */
+public class GeoWebCacheTypedEnvironment {
+
+    private final GeoWebCacheEnvironment environment;
+
+    public GeoWebCacheTypedEnvironment(GeoWebCacheEnvironment environment) {
+        this.environment = environment;
+    }
+
+    private String resolve(String value) {
+        if (GeoWebCacheEnvironment.ALLOW_ENV_PARAMETRIZATION)
+            return (String) environment.resolveValue(value);
+        else return value;
+    }
+
+    /**
+     * Resolve placeholders for a String parameter if enabled on configurations.
+     *
+     * @param value String value with optional placeholders.
+     * @param type Data Type to return
+     * @return Optional resolved value.
+     */
+    public <T extends Object> Optional<T> resolve(final String value, Class<T> type) {
```

Comment:
I'd keep method name conventions intact, call it ``resolveValue`` like in ``GeoWebCacheEnvironment`` and it has the extra parameter for type conversion.

Reply:
Added as resolveValueIfEnabled since this method does the check for parameter-replacement flag enabled inside (the former 'resolveValue' method doesn't).